packages/react-avatar/src/components/Avatar/Avatar.types.ts

```diff
@@ -1,99 +1,105 @@
 import * as React from 'react';
-import type { ComponentPropsCompat, ComponentStateCompat, ShorthandPropsCompat } from '@fluentui/react-utilities';
-import type { PresenceBadgeProps, PresenceBadgeStatus } from '@fluentui/react-badge';
+import type {
+  ComponentProps,
+  ComponentState,
+  IntrinsicShorthandProps,
+  ObjectShorthandProps,
+} from '@fluentui/react-utilities';
+import type { PresenceBadgeProps } from '@fluentui/react-badge';
 
-export type AvatarProps = ComponentPropsCompat &
-  React.HTMLAttributes<HTMLElement> & {
-    /**
-     * The Avatar's image.
-     */
-    image?: ShorthandPropsCompat<React.ImgHTMLAttributes<HTMLImageElement>>;
+export type AvatarSlots = {
+  root: Omit<IntrinsicShorthandProps<'span'>, 'color'> & { children?: never };
 
-    /**
-     * The label shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
-     */
-    label?: ShorthandPropsCompat<React.HTMLAttributes<HTMLElement>>;
+  /**
+   * The Avatar's image.
+   */
+  image?: IntrinsicShorthandProps<'img'> & { children?: never };
```

Comment:
⚠ It's not correct as then you cannot use `children` as a function:

```tsx
<Avatar image={{ (Component, props) => {} }} />
```

Reply:
Fixed in [439ce8b](https://github.com/microsoft/fluentui/pull/19449/commits/439ce8b5b48cc3e1826afd45386863a04878102e)